azurerm/internal/services/containers/kubernetes_nodepool.go

```diff
@@ -121,6 +121,16 @@ func SchemaDefaultNodePool() *schema.Schema {
 					ValidateFunc: validation.IntAtLeast(1),
 				},
 
+				"os_disk_type": {
+					Type:     schema.TypeString,
+					Optional: true,
+					ForceNew: true,
+					ValidateFunc: validation.StringInSlice([]string{
+						"Ephemeral",
+						"Managed",
```

Comment:
there's constants for these in the SDK - could we use these:

```
// OSDiskType enumerates the values for os disk type.
type OSDiskType string

const (
	// Ephemeral ...
	Ephemeral OSDiskType = "Ephemeral"
	// Managed ...
	Managed OSDiskType = "Managed"
)
```

```suggestion
						string(containerservice.Ephemeral),
						string(containerservice.Managed),
```